src/poetry/core/version/pep440/version.py

```diff
@@ -201,46 +202,66 @@ def is_unstable(self) -> bool:
     def is_stable(self) -> bool:
         return not self.is_unstable()
 
+    def _is_increment_required(self) -> bool:
+        return self.is_stable() or (not self.is_prerelease() and self.is_postrelease())
+
     def next_major(self: T) -> T:
         release = self.release
-        if self.is_stable() or Release(self.release.major, 0, 0) < self.release:
+        if (
+            self._is_increment_required()
+            or Release(self.release.major, 0, 0) < self.release
+        ):
             release = self.release.next_major()
         return self.__class__(epoch=self.epoch, release=release)
 
     def next_minor(self: T) -> T:
         release = self.release
         if (
-            self.is_stable()
+            self._is_increment_required()
             or Release(self.release.major, self.release.minor, 0) < self.release
         ):
             release = self.release.next_minor()
         return self.__class__(epoch=self.epoch, release=release)
 
     def next_patch(self: T) -> T:
-        return self.__class__(
-            epoch=self.epoch,
-            release=self.release.next_patch() if self.is_stable() else self.release,
-        )
+        release = self.release
```

Comment:
Any reason to define this variable? Seems like it should either be used in the condition below instead of `self.release`, or not be defined at all.

Reply:
It's just copied from the other methods. It is required (at least in the `else` case), but you're right that it can be used instead of of `self.release` in the condition. I'll change this.